runtime/compiler/codegen/J9AheadOfTimeCompile.cpp

```diff
@@ -240,6 +243,87 @@ J9::AheadOfTimeCompile::initializeCommonAOTRelocationHeader(TR::IteratedExternal
          }
          break;
 
+      case TR_CheckMethodEnter:
+         {
+         TR_RelocationRecordMethodEnterCheck *mcRecord = reinterpret_cast<TR_RelocationRecordMethodEnterCheck *>(reloRecord);
+
+         mcRecord->setDestinationAddress(reloTarget, reinterpret_cast<uintptrj_t>(relocation->getTargetAddress()));
+         }
+         break;
+
+      case TR_VerifyClassObjectForAlloc:
+         {
+         TR_RelocationRecordVerifyClassObjectForAlloc *allocRecord = reinterpret_cast<TR_RelocationRecordVerifyClassObjectForAlloc *>(reloRecord);
+
+         TR::SymbolReference * classSymRef = reinterpret_cast<TR::SymbolReference *>(relocation->getTargetAddress());
+         TR_RelocationRecordInformation *recordInfo = reinterpret_cast<TR_RelocationRecordInformation*>(relocation->getTargetAddress2());
+         TR::LabelSymbol *label = reinterpret_cast<TR::LabelSymbol *>(recordInfo->data3);
+         TR::Instruction *instr = reinterpret_cast<TR::Instruction *>(recordInfo->data4);
+
+         uint32_t branchOffset = static_cast<uint32_t>(label->getCodeLocation() - instr->getBinaryEncoding());
+
+         allocRecord->setInlinedSiteIndex(reloTarget, static_cast<uintptrj_t>(recordInfo->data2));
+         allocRecord->setConstantPool(reloTarget, reinterpret_cast<uintptrj_t>(classSymRef->getOwningMethod(comp)->constantPool()));
+         allocRecord->setBranchOffset(reloTarget, static_cast<uintptrj_t>(branchOffset));
+         allocRecord->setAllocationSize(reloTarget, static_cast<uintptrj_t>(recordInfo->data1));
+
+         /* Temporary, will be cleaned up in a future PR */
+         if (comp->getOption(TR_UseSymbolValidationManager))
```

Comment:
This code wasn't consistently being applied on all platforms (e.g. on ARM there only a `cpIndex` was being written). I'm ok with making this change as part of the consolidation  as part of its "value", but please add a note in the appropriate commit messages that this change occurs.